src/check_release.rs

```diff
@@ -482,115 +536,211 @@ mod test {
-        let actual = classify_semver_version_change(baseline, current);
+        let expected = Some(VersionChange {
+            level: ActualSemverUpdate::Minor,
+            kind: VersionChangeKind::Actual,
+        });
+        let actual = classify_minimum_semver_version_change(baseline, current);
         assert_eq!(actual, expected);
     }
 
     #[test]
     fn classify_major_changed() {
         let baseline = Some("0.9.0");
         let current = Some("1.0.0");
-        let expected = Some(ActualSemverUpdate::Major);
-        let actual = classify_semver_version_change(baseline, current);
+        let expected = Some(VersionChange {
+            level: ActualSemverUpdate::Major,
+            kind: VersionChangeKind::Actual,
+        });
+        let actual = classify_minimum_semver_version_change(baseline, current);
         assert_eq!(actual, expected);
     }
 
     #[test]
     fn classify_zerover_minor_changed() {
         let baseline = Some("0.1.0");
         let current = Some("0.1.1");
-        let expected = Some(ActualSemverUpdate::Minor);
-        let actual = classify_semver_version_change(baseline, current);
+        let expected = Some(VersionChange {
+            level: ActualSemverUpdate::Minor,
+            kind: VersionChangeKind::Actual,
+        });
+        let actual = classify_minimum_semver_version_change(baseline, current);
         assert_eq!(actual, expected);
     }
 
     #[test]
     fn classify_zerover_major_changed() {
         let baseline = Some("0.1.0");
         let current = Some("0.2.0");
-        let expected = Some(ActualSemverUpdate::Major);
-        let actual = classify_semver_version_change(baseline, current);
+        let expected = Some(VersionChange {
+            level: ActualSemverUpdate::Major,
+            kind: VersionChangeKind::Actual,
+        });
+        let actual = classify_minimum_semver_version_change(baseline, current);
         assert_eq!(actual, expected);
     }
 
     #[test]
     fn classify_double_zerover_major_changed() {
         let baseline = Some("0.0.1");
         let current = Some("0.0.2");
-        let expected = Some(ActualSemverUpdate::Major);
-        let actual = classify_semver_version_change(baseline, current);
+        let expected = Some(VersionChange {
+            level: ActualSemverUpdate::Major,
+            kind: VersionChangeKind::Actual,
+        });
+        let actual = classify_minimum_semver_version_change(baseline, current);
         assert_eq!(actual, expected);
     }
 
     #[test]
     fn classify_pre_same() {
         let baseline = Some("1.0.0-alpha.0");
         let current = Some("1.0.0-alpha.0");
-        let expected = Some(ActualSemverUpdate::NotChanged);
-        let actual = classify_semver_version_change(baseline, current);
+        let expected = Some(VersionChange {
+            level: ActualSemverUpdate::Patch,
+            kind: VersionChangeKind::Minimum,
+        });
```

Comment:
I think this might be incorrect. Compare this to lines 643-646 below: bumping any component of a pre-release is major relative to that pre-release (but not relative to other stable releases in that range, see lines around 667).

Reply:
yeah I think so! When we are in a pre-release version, the minimum possible change would be a major, not a patch.